share/eds/eds.go

```diff
@@ -273,3 +277,59 @@ func ReadEDS(ctx context.Context, r io.Reader, root share.DataHash) (eds *rsmt2d
 	}
 	return eds, nil
 }
+
+// ConstructFn aliases a function that creates an ExtendedDataSquare.
+type ConstructFn func(
+	data [][]byte,
+	appVersion uint64,
+	squareSizeBounds int,
+	options ...nmt.Option,
+) (*rsmt2d.ExtendedDataSquare, error)
+
+// MakeExtendedDataSquare extends the given block data, returning the resulting
+// ExtendedDataSquare (EDS). If the data is empty,
+// nil is returned in place of the eds.
+func MakeExtendedDataSquare(
+	data [][]byte,
+	appVersion uint64,
+	squareSizeBounds int,
+	options ...nmt.Option,
+) (*rsmt2d.ExtendedDataSquare, error) {
+	if len(data) == 0 {
+		return nil, nil
+	}
+
+	// Construct the data square from the block's transactions
+	dataSquare, err := square.Construct(data, appVersion, squareSizeBounds)
+	if err != nil {
+		return nil, err
+	}
+	return extendShares(shares.ToBytes(dataSquare), options...)
+}
+
+func extendShares(s [][]byte, options ...nmt.Option) (*rsmt2d.ExtendedDataSquare, error) {
+	// Check that the length of the square is a power of 2.
+	if !shares.IsPowerOfTwo(len(s)) {
+		return nil, fmt.Errorf("number of shares is not a power of 2: got %d", len(s))
+	}
+	// here we construct a tree
+	// Note: uses the nmt wrapper to construct the tree.
+	squareSize := square.Size(len(s))
+	return rsmt2d.ComputeExtendedDataSquare(s,
+		appconsts.DefaultCodec(),
+		wrapper.NewConstructor(uint64(squareSize),
+			options...))
```

Comment:
this has weird formatting, im surprised go linter accepts it, could you clean it up a bit?